service/src/main/java/bio/terra/workspace/service/resource/controlled/ControlledResourceService.java

```diff
@@ -480,6 +483,32 @@ public ControlledResource getControlledResource(UUID workspaceUuid, UUID resourc
     return resourceDao.getResource(workspaceUuid, resourceId).castToControlledResource();
   }
 
+  /**
+   * Synchronously delete all controlled resources instances of the specified type returning all
+   * remaining resources
+   *
+   * @param workspaceUuid workspace UUID
+   * @param allResources list of all controlled resources in the workspace
+   * @param type resource type to delete
+   * @param userRequest the user request
+   * @return list of controlled resources, after deleting those of specified type
+   */
+  public @NotNull List<ControlledResource> deleteControlledResourceSyncOfType(
```

Comment:
nit: rm @notnull. generally we mark nullable and not the other way around.

Reply:
Reason for this - 
specifying NotNull has 2 uses here - 1) compiler can enforce that null is never returned. 2) caller can be assured that this will never be null and hence avoid checks

Reply:
https://engdoc.corp.google.com/eng/doc/devguide/java/practices/null.md#legacy

Reply:
Done, all places